client/src/lib/hooks/useConnection.ts

```diff
@@ -224,6 +224,19 @@ export function useConnection({
         await client.connect(clientTransport);
       } catch (error) {
         console.error("Failed to connect to MCP server:", error);
+
+        // Extract detailed error message from SSE error response when we use stdio transport
+        if (error instanceof SseError && transportType === "stdio") {
+          try {
+            const response = await fetch(backendUrl.toString(), { headers });
```

Comment:
I think we shouldn't need to try the connection again here. If we want additional information, we should extract it from the `SseError` ideally. If the `SseError` is missing information, we could try adding more from the inspector server, or from the SDK (depending on what's dropping info).